#### This sheet is for cleaning and EDA on CALIFORNIA data

Gameplan to get input for rough model
1. location
1. covid
1. health
1. govt action

Cleaning steps
1. nulls
1. datatypes
1. ??

EDA steps

In [2]:
# imports

import pandas as pd

In [3]:
# other settings
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

In [5]:
# read in CA data
data = pd.read_csv('./data/data_ca_raw.csv')
data.shape

/Users/nevochka/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3071: DtypeWarning: Columns (12,13) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


(17700, 110)

In [8]:
data.info(4)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17700 entries, 0 to 17699
Data columns (total 110 columns):
 #   Column                              Dtype  
---  ------                              -----  
 0   Unnamed: 0                          int64  
 1   Unnamed: 0.1                        int64  
 2   key                                 object 
 3   date                                object 
 4   wikidata                            object 
 5   datacommons                         object 
 6   country_code                        object 
 7   country_name                        object 
 8   subregion1_code                     object 
 9   subregion1_name                     object 
 10  subregion2_code                     float64
 11  subregion2_name                     object 
 12  locality_code                       object 
 13  locality_name                       object 
 14  3166-1-alpha-2                      object 
 15  3166-1-alpha-3                      object 
 16  agg

In [44]:
data['subregion2_name'].value_counts()

Humboldt County           295
Stanislaus County         295
Placer County             295
Napa County               295
Inyo County               295
Riverside County          295
Yuba County               295
Modoc County              295
Mendocino County          295
Glenn County              295
San Bernardino County     295
Tuolumne County           295
Colusa County             295
Kings County              295
San Mateo County          295
San Luis Obispo County    295
Orange County             295
Monterey County           295
San Francisco County      295
Los Angeles County        295
Lassen County             295
Calaveras County          295
Kern County               295
Solano County             295
Amador County             295
Alameda County            295
Alpine County             295
Sonoma County             295
Mariposa County           295
Merced County             295
Sutter County             295
Siskiyou County           295
Santa Clara County        295
Shasta Cou

In [14]:
# isolate location info
# new = old[['A', 'C', 'D']].copy()

data_geo = data[['open_street_maps',
 'latitude',
 'longitude',
 'elevation',
 'area',
 'rural_area',
 'urban_area']].copy()

In [15]:
data_geo.shape

(17700, 7)

In [20]:
data_geo.head()

open_street_maps  latitude  longitude  elevation      area  rural_area  \
0          165475.0      37.0     -120.0      884.0  423970.0         NaN   
1          165475.0      37.0     -120.0      884.0  423970.0         NaN   
2          165475.0      37.0     -120.0      884.0  423970.0         NaN   
3          165475.0      37.0     -120.0      884.0  423970.0         NaN   
4          165475.0      37.0     -120.0      884.0  423970.0         NaN   

   urban_area  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN

In [18]:
# save to file for each state
data_geo.to_csv('./data/data_ca_geo.csv')
# data_geo.to_csv('./data/data_colo_geo.csv')
# data_geo.to_csv('./data/data_oh_geo.csv')

In [21]:
data_geo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17700 entries, 0 to 17699
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   open_street_maps  17700 non-null  float64
 1   latitude          17700 non-null  float64
 2   longitude         17700 non-null  float64
 3   elevation         1180 non-null   float64
 4   area              17700 non-null  float64
 5   rural_area        0 non-null      float64
 6   urban_area        0 non-null      float64
dtypes: float64(7)
memory usage: 968.1 KB


In [26]:
data_geo['latitude'].value_counts()

39.590000    590
37.930000    590
38.270000    295
38.210000    295
39.430000    295
37.777500    295
37.000000    295
40.130000    295
39.270000    295
34.830000    295
36.750000    295
33.020000    295
36.583333    295
40.660000    295
38.580000    295
37.220000    295
34.360000    295
38.500000    295
37.190000    295
41.740000    295
37.030000    295
36.610000    295
36.070000    295
38.553889    295
33.730000    295
39.040000    295
39.300000    295
40.650000    295
37.440000    295
38.040000    295
40.760000    295
37.916667    295
37.778333    295
39.090000    295
36.240000    295
40.010000    295
33.670000    295
35.380000    295
36.230000    295
38.020000    295
41.583333    295
40.800000    295
35.340000    295
39.660000    295
39.060000    295
38.440000    295
37.580000    295
38.780000    295
34.540000    295
37.360000    295
38.450000    295
34.050000    295
38.510000    295
39.180000    295
41.600000    295
37.650000    295
37.560000    295
33.040000    295
Name: latitude

#### Geographic info -- summary:
https://github.com/GoogleCloudPlatform/covid-19-open-data/blob/main/docs/table-geography.md

California:
1. There are 295 observations (rows) for CA-state, and for each county (58 counties)
1. Observations are at the day level (Oct 2, Oct 3, etc.), starting Jan 1, 2020
1. No data for:  elevation, rural vs. urban designation
1. All geo values are floats.

Colorado, Ohio -- TBD

In [27]:
# create a lat-long file
# new = old[['A', 'C', 'D']].copy()

data_latlong = data[['latitude', 'longitude']].copy()
data_latlong.shape

In [29]:
data_latlong.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17700 entries, 0 to 17699
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   latitude   17700 non-null  float64
 1   longitude  17700 non-null  float64
dtypes: float64(2)
memory usage: 276.7 KB


In [30]:
data_latlong.head()

latitude  longitude
0      37.0     -120.0
1      37.0     -120.0
2      37.0     -120.0
3      37.0     -120.0
4      37.0     -120.0

In [31]:
# save latlong data to file for each state
data_latlong.to_csv('./data/data_ca_latlong.csv')
# data_latlong.to_csv('./data/data_colo_latlong.csv')
# data_latlong.to_csv('./data/data_oh_latlong.csv')

### Covid info -- summary:
https://github.com/GoogleCloudPlatform/covid-19-open-data/blob/main/docs/table-epidemiology.md

California:
1. The columns with the most Covid data are 
> new_confirmed, new_recovered, total_confirmed, total_deceased.
1. Covid data has several empty columns, include hospitalizations, intensive care, ventilators
    
    
Colorado, Ohio -- TBD

In [45]:
data_covid = data[['date', 'new_confirmed',
 'new_deceased',
 'new_recovered',
 'new_tested',
 'total_confirmed',
 'total_deceased',
 'total_recovered',
 'total_tested',
 'new_hospitalized',
 'total_hospitalized',
 'current_hospitalized',
 'new_intensive_care',
 'total_intensive_care',
 'current_intensive_care',
 'new_ventilator',
 'total_ventilator',
 'current_ventilator']].copy()
data_covid.shape

(17700, 18)

In [46]:
data_covid.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17700 entries, 0 to 17699
Data columns (total 18 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   date                    17700 non-null  object 
 1   new_confirmed           12994 non-null  float64
 2   new_deceased            12768 non-null  float64
 3   new_recovered           0 non-null      float64
 4   new_tested              461 non-null    float64
 5   total_confirmed         13059 non-null  float64
 6   total_deceased          12827 non-null  float64
 7   total_recovered         0 non-null      float64
 8   total_tested            462 non-null    float64
 9   new_hospitalized        0 non-null      float64
 10  total_hospitalized      0 non-null      float64
 11  current_hospitalized    417 non-null    float64
 12  new_intensive_care      0 non-null      float64
 13  total_intensive_care    0 non-null      float64
 14  current_intensive_care  417 non-null  

In [39]:
data_covid.head().T

0   1   2   3   4
new_confirmed          NaN NaN NaN NaN NaN
new_deceased           NaN NaN NaN NaN NaN
new_recovered          NaN NaN NaN NaN NaN
new_tested             NaN NaN NaN NaN NaN
total_confirmed        NaN NaN NaN NaN NaN
total_deceased         NaN NaN NaN NaN NaN
total_recovered        NaN NaN NaN NaN NaN
total_tested           NaN NaN NaN NaN NaN
new_hospitalized       NaN NaN NaN NaN NaN
total_hospitalized     NaN NaN NaN NaN NaN
current_hospitalized   NaN NaN NaN NaN NaN
new_intensive_care     NaN NaN NaN NaN NaN
total_intensive_care   NaN NaN NaN NaN NaN
current_intensive_care NaN NaN NaN NaN NaN
new_ventilator         NaN NaN NaN NaN NaN
total_ventilator       NaN NaN NaN NaN NaN
current_ventilator     NaN NaN NaN NaN NaN

In [47]:
# save Covid data to file for each state

data_covid.to_csv('./data/data_ca_covid.csv')
# data_covid.to_csv('./data/data_colo_covid.csv')
# data_covid.to_csv('./data/data_oh_covid.csv')